### PR TITLE
Feature/filters

### PR DESCRIPTION
**Pull request by Alex-Kalanda**
I'm sorry) the PR is not small, but all this code is interconnected

### CODE REVIEW

File: src/components/BasketProduct/BasketProduct.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ class BasketProduct {
       <td class="basket-product__index">
         <span>${index}</span>
       </td>
-      <td class="basket-product__info" onclick="window.location.pathname = '/product/${id}'">
+      <td class="basket-product__info" onclick="window.location.href = '/product/${id}'">
```
Comment (Ksarelto) [inline src/components/BasketProduct/BasketProduct.ts:26]: is u click on this td window will reload page is it ok?

File: src/utils/UrlFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      const [minPrice, maxPrice] = temp.price.split(VALUE_SEPARATOR);
+      this.objQueryParams.price = [+minPrice, +maxPrice];
+    }
+
+    if (temp.stock) {
+      const [minStock, maxStock] = temp.stock.split(VALUE_SEPARATOR);
+      this.objQueryParams.stock = [+minStock, +maxStock];
+    }
+
+    this.objQueryParams.mode = temp.mode || '';
```
Comment (Ksarelto) [inline src/utils/UrlFormatter.ts:47]:
```
this.objQueryPrams = {
  ...this.objQueryParams,
 mode: temp.mode || '',
 sort: temp.sort || ''
}
```

File: src/components/FilterList/FilterList.style.scss [hunk 3]
```diff
@@ -18,26 +18,48 @@
...
 
   &__item {
     font-size: 1.6rem;
     font-weight: 400;
+    display: flex;
+    align-items: center;
 
     &__input {
-      display: none;
+      /* display: none; */
```
Comment (Ksarelto) [inline src/components/FilterList/FilterList.style.scss:37]: why it is here?

File: src/components/containers/Sidebar/Sidebar.ts [hunk 4]
```diff
@@ -1,48 +1,60 @@
...
+    const priceOptions: IRangeOptions = {
+      rangeTitle: FilterParamTitles.price,
+      scaleLimits: data.priceScale,
+      currentPoses: data.price,
+      cbRender: this.cbRender,
+      symbolsAfterComma: 2,
+      step: 0.01,
+      symbol: Symbol.CURRENCY,
+    };
+    const stockOptions: IRangeOptions = {
```
Comment (Ksarelto) [inline src/components/containers/Sidebar/Sidebar.ts:41]: add empty line between this two big objects and before 'return'

File: src/components/ProductsGrid/ProductsGrid.ts [hunk 5]
```diff
@@ -1,15 +1,30 @@
...
+  constructor(private data: IProcessedData) {
+    this.data = data;
+    this.productsDataList = [...this.data.productsId].map((id: number) => Database.getProductById(id));
+  }
 
   render(): string {
-    const productsList = this.productsDataList.reduce((acc: string, data: IProduct) => acc + new Product(data).render(), '');
+    const sortedDataList = this.getSortedList(this.data.sort);
+    const productsList = sortedDataList.reduce((acc: string, data: IProduct) => acc + new Product(data).render(), '');
+    const goodsNotFind = `<span class="grid__empty">${NO_PRODUCT_TEXT}</span>`;
```
Comment (Ksarelto) [inline src/components/ProductsGrid/ProductsGrid.ts:21]: notFound ))

File: src/components/Controls/Controls.ts [hunk 6]
```diff
@@ -3,79 +3,133 @@ import modeGrid3Icon from '../../assets/svg/grid3_icon.svg';
...
 import searchIcon from '../../assets/svg/search_icon.svg';
 import searchCross from '../../assets/svg/search_cross.svg';
+import { IProcessedData } from '../../database/DataBase.interfaces';
+import Handler from '../../utils/Handler';
+import { Events } from '../../common.types/enums';
+import { findElem } from '../../utils/findElem';
+import UrlFormatter, { QueryNames } from '../../utils/UrlFormatter';
+import { DEFAULT_MODE } from '../ProductsGrid/ProductsGrid.const';
+import {
+  SEARCH_PLACEHOLDER, SEARCH_RESULT_TITLE, SORT_TITLE, SortNames,
```
Comment (Ksarelto) [inline src/components/Controls/Controls.ts:13]: if u use many imports better to set them in column
```
import {
   SEARCH_PLACEHOLDER,
   SEARCH_RESULT_TITLE,
   SORT_TITLE,
   SortNames,
} from './Controls.const';
```

File: src/components/Controls/Controls.ts [hunk 7]
```diff
@@ -3,79 +3,133 @@ import modeGrid3Icon from '../../assets/svg/grid3_icon.svg';
...
     return `
       <div class="controls__sort sorter">
         <span class="sorter__title">${SORT_TITLE}</span>
         <div class="sorter__droplist-wrap">
           <select class="sorter__droplist" name="sort" id="sorter-droplist">
-            <option value="upDiscount">${SortingNames.discount.fromLowToHigh}</option>
-            <option value="downDiscount">${SortingNames.discount.fromHighToLow}</option>
-            <option value="upStock">${SortingNames.stock.fromLowToHigh}</option>
-            <option value="downStock">${SortingNames.stock.fromHighToLow}</option>
+            /* <option ${SortValues.none === sortType ? 'selected' : ''} value=${SortValues.none}>
```
Comment (Ksarelto) [inline src/components/Controls/Controls.ts:41]: why is it commented?

File: src/components/Controls/Controls.ts [hunk 8]
```diff
@@ -3,79 +3,133 @@ import modeGrid3Icon from '../../assets/svg/grid3_icon.svg';
...
+  getSearchBlock(amount: number, search: string): string {
     return `
       <div class="controls__search search">
         <div class="search__result">
           <span class="search__result__title">${SEARCH_RESULT_TITLE}</span>
-          <span class="search__result__amount" id="search-amount">56</span>
+          <span class="search__result__amount" id="search-amount">${amount}</span>
         </div>
         <div class="search__input-wrap">
           <input type="text" class="search__input" id="search-input" placeholder=${SEARCH_PLACEHOLDER}
```
Comment (Ksarelto) [inline src/components/Controls/Controls.ts:69]: if some tag has a lot of attributes, set them in column
```
<input
   type="text"
   class="search__input"
   id="search-input"
   placeholder=${SEARCH_PLACEHOLDER}
   autofocus
   autocomplete="off"
   value=${search}
>
```

File: src/components/Controls/Controls.ts [hunk 9]
```diff
@@ -3,79 +3,133 @@ import modeGrid3Icon from '../../assets/svg/grid3_icon.svg';
...
-        <div class="controls__mode__item ${value === ViewMode.grid4 ? 'active' : ''}">
+        <div class="controls__mode__item ${mode === ViewMode.grid4 ? 'active' : ''}" 
+            id="switch-${ViewMode.grid4}-btn" data-view-mode=${ViewMode.grid4}>
           <img src=${modeGrid4Icon} alt="icon mode">
         </div>
       </div>`;
   }
+
+  setHandlers(mode: string): void {
+    // search button
```
Comment (Ksarelto) [inline src/components/Controls/Controls.ts:97]: ?

Comment (Ksarelto) [reply src/components/Controls/Controls.ts:97]: it is not ok, if u write comments to describe what your methods do

File: src/components/BasketProduct/BasketProduct.ts (hunk 1 above)
Comment (Alex-Kalanda) [reply src/components/BasketProduct/BasketProduct.ts:26]: We decided to do it in this place in such way that there are no hashes

File: src/utils/UrlFormatter.ts (hunk 2 above)
Comment (Alex-Kalanda) [reply src/utils/UrlFormatter.ts:47]: ok

File: src/components/FilterList/FilterList.style.scss (hunk 3 above)
Comment (Alex-Kalanda) [reply src/components/FilterList/FilterList.style.scss:37]: oops! its trash)

File: src/components/containers/Sidebar/Sidebar.ts (hunk 4 above)
Comment (Alex-Kalanda) [reply src/components/containers/Sidebar/Sidebar.ts:41]: ok

File: src/components/ProductsGrid/ProductsGrid.ts (hunk 5 above)
Comment (Alex-Kalanda) [reply src/components/ProductsGrid/ProductsGrid.ts:21]: corrected)

File: src/components/Controls/Controls.ts (hunk 6 above)
Comment (Alex-Kalanda) [reply src/components/Controls/Controls.ts:13]: ok

File: src/components/Controls/Controls.ts (hunk 7 above)
Comment (Alex-Kalanda) [reply src/components/Controls/Controls.ts:41]: hmmm.. deleted

File: src/components/Controls/Controls.ts (hunk 8 above)
Comment (Alex-Kalanda) [reply src/components/Controls/Controls.ts:69]: ok, clear

File: src/components/Controls/Controls.ts (hunk 9 above)
Comment (Alex-Kalanda) [reply src/components/Controls/Controls.ts:97]: even so small? 
i've done

**Pull request merged**